lib/ansible/plugins/connection/__init__.py

```diff
@@ -254,10 +254,8 @@ def check_password_prompt(self, b_output):
             return False
         elif isinstance(self._play_context.prompt, string_types):
             b_prompt = to_bytes(self._play_context.prompt).strip()
-            b_lines = b_output.splitlines(True)
-            if not b_lines:
-                return False
-            return b_lines[-1].strip().endswith(b_prompt) or b_lines[0].strip().endswith(b_prompt)
+            b_lines = b_output.splitlines()
+            return any(l.strip().startswith(b_prompt) for l in b_lines)
```

Comment:
I suppose I could, but not sure what that would get us.  We're going to test them all anyway, and if any are True it will pass.

Before we did: `line[0] or line[-1]`

This just ensures that if there were more than 2 lines, it would still check them all.